cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java

```diff
@@ -175,6 +177,92 @@ public String cruiseControlHost(String clusterName, String clusterNamespace) {
+        this.testStateWithWrongAnnotation(vertx, context, 0, CruiseControlEndpoints.REMOVE_BROKER, kcRebalance, KafkaRebalanceState.ProposalReady, KafkaRebalanceAnnotation.unknown);
+    }
+
+    @Test
+    public void testReadyWithWrongAnnotation(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+        KafkaRebalance kcRebalance = createKafkaRebalance(KafkaRebalanceState.Ready, null, null, EMPTY_KAFKA_REBALANCE_SPEC);
+        this.testStateWithWrongAnnotation(vertx, context, 0, CruiseControlEndpoints.REBALANCE, kcRebalance, KafkaRebalanceState.Ready, KafkaRebalanceAnnotation.approve);
+
+        kcRebalance = createKafkaRebalance(KafkaRebalanceState.Ready, null, null, ADD_BROKER_KAFKA_REBALANCE_SPEC);
+        this.testStateWithWrongAnnotation(vertx, context, 0, CruiseControlEndpoints.ADD_BROKER, kcRebalance, KafkaRebalanceState.Ready, KafkaRebalanceAnnotation.approve);
+
+        kcRebalance = createKafkaRebalance(KafkaRebalanceState.Ready, null, null, REMOVE_BROKER_KAFKA_REBALANCE_SPEC);
+        this.testStateWithWrongAnnotation(vertx, context, 0, CruiseControlEndpoints.REMOVE_BROKER, kcRebalance, KafkaRebalanceState.Ready, KafkaRebalanceAnnotation.approve);
+    }
+
+    private void testStateWithWrongAnnotation(Vertx vertx, VertxTestContext context,
+                                              int pendingCalls, CruiseControlEndpoints endpoint,
+                                              KafkaRebalance kcRebalance, KafkaRebalanceState kafkaRebalanceState,
+                                              KafkaRebalanceAnnotation kafkaRebalanceAnnotation) throws IOException, URISyntaxException {
+        MockCruiseControl.setupCCRebalanceResponse(ccServer, pendingCalls, endpoint);
+
+        checkTransitionWithWrongAnnotation(vertx, context,
+                kafkaRebalanceState, kafkaRebalanceState,
+                kafkaRebalanceAnnotation, kcRebalance)
+                .onComplete(result -> defaultStatusHandler(result, context));
+    }
+
+    /**
+     *  Checks the expected transition between two states of the Kafka Rebalance operator when wrong annotation is applied.
+     *
+     * @param vertx The vertx test instance.
+     * @param context The test context instance.
+     * @param currentState The current state of the resource before being passed to computeNextStatus.
+     * @param nextState The expected state of the resouce after computeNextStatus has been called.
+     * @param initialAnnotation The initial annotation attached to the Kafka Rebalance resource. For example none or refresh.
+     * @param kcRebalance The Kafka Rebalance instance that will be returned by the resourceSupplier.
+     * @return A future for the {@link KafkaRebalanceStatus} returned by the {@link KafkaRebalanceAssemblyOperator#computeNextStatus} method
+     */
+
+    private Future<KafkaRebalanceStatus> checkTransitionWithWrongAnnotation(Vertx vertx, VertxTestContext context,
```

Comment:
Almost all of this method is identitcal to the other checkTransition method? Can you either alter that method to take an annotation option or extract the common code to another method?